text/0000-ordered-queries.md

```diff
@@ -0,0 +1,126 @@
+- Feature Name: ordered-queries
+- Start Date: 2015-09-25
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add the following to BTreeMap
+
+* pred_inc
+* pred_exc
+* succ_inc
+* succ_exc
+* first
+* last
+* pred_inc_mut
+* pred_exc_mut
+* succ_inc_mut
+* succ_exc_mut
+* first_mut
+* last_mut
+
+and to BTreeSet:
+
+* pred_inc
+* pred_exc
+* succ_inc
+* succ_exc
+* first
+* last
+
```

Comment:
Incidentally the lack of genericity over mutability is killing me. Don't how I'd do it, but there's _so_ much repetition in API's these days because of it.